auto_dev/cli_executor.py

```diff
@@ -88,6 +88,10 @@ def _execute_stream(self, verbose: bool = True, shell: bool = False) -> bool | N
                         logger.error("Command failed with return code: %s", self.return_code)
                     return False
                 return True
+        except KeyboardInterrupt:
+            logger.info("Command execution interrupted by user.")
+            process.terminate()
```

Comment:
This way we don't leave dangling processes behind and can re-run the script without port conflicts.